File: sendBack/src/main/java/com/logistics/entity/Order.java
package com.logistics.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("t_order")
@Schema(description = "订单实体")
public class Order {

    @TableId(type = IdType.AUTO)
    @Schema(description = "订单ID")
    private Long id;

    @Schema(description = "快递单号")
    private String trackingNumber;

    @Schema(description = "快递公司")
    private String courierCompany;

    @Schema(description = "快递公司Logo")
    private String courierLogo;

    @Schema(description = "订单状态(pending-待取件,in_transit-运输中,delivered-已签收,exception-异常)")
    private String status;

    @Schema(description = "物品名称")
    private String itemName;

    @Schema(description = "寄件人姓名")
    private String senderName;

    @Schema(description = "寄件人电话")
    private String senderPhone;

    @Schema(description = "寄件人地址")
    private String senderAddress;

    @Schema(description = "收件人姓名")
    private String receiverName;

    @Schema(description = "收件人电话")
    private String receiverPhone;

    @Schema(description = "收件人地址")
    private String receiverAddress;

    @Schema(description = "预计送达时间")
    private String estimatedTime;

    @Schema(description = "用户ID")
    private Long userId;

    @TableField(fill = FieldFill.INSERT)
    @Schema(description = "创建时间")
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    @Schema(description = "更新时间")
    private LocalDateTime updatedAt;

    @TableLogic
    @Schema(description = "是否删除")
    private Integer deleted;
}

File: sendFront/src/api/order.ts
import { del, get, post, put } from './request'

// 订单状态
export type OrderStatus = 'pending' | 'in_transit' | 'delivered' | 'exception'

// 联系人信息
export interface ContactInfo {
  name: string
  phone: string
  address: string
}

// 创建订单请求参数（与后端DTO对应）
export interface OrderCreateDTO {
  courierCompany: string
  courierLogo?: string
  itemName: string
  senderName: string
  senderPhone: string
  senderAddress: string
  receiverName: string
  receiverPhone: string
  receiverAddress: string
  estimatedTime?: string
}

// 订单查询参数（与后端DTO对应）
export interface OrderQueryDTO {
  status?: OrderStatus
  keyword?: string
  page?: number
  pageSize?: number
}

// 订单响应
export interface OrderVO {
  id: number
  trackingNumber: string
  courierCompany: string
  courierLogo: string | null
  status: OrderStatus
  statusText: string
  itemName: string
  senderName: string
  senderPhone: string
  senderAddress: string
  receiverName: string
  receiverPhone: string
  receiverAddress: string
  estimatedTime: string | null
  createdAt: string
  logisticsNodes?: LogisticsNodeVO[]
}

// 物流节点响应（嵌入订单中）
export interface LogisticsNodeVO {
  id: number
  orderId: number
  trackingNumber: string
  time: string
  location: string
  status: string
  description: string
  isActive: boolean
}

// 分页响应（与后端一致）
export interface PageVO<T> {
  list: T[]
  total: number
  totalPages: number
  page: number
  pageSize: number
}

// 获取订单列表
export function getOrderList(params?: OrderQueryDTO) {
  return get<PageVO<OrderVO>>('/orders', { params })
}

// 获取订单详情
export function getOrderById(id: number) {
  return get<OrderVO>(`/orders/${id}`)
}

// 根据快递单号查询订单
export function getOrderByTrackingNumber(trackingNumber: string) {
  return get<OrderVO>(`/orders/tracking/${trackingNumber}`)
}

// 创建订单
export function createOrder(data: OrderCreateDTO) {
  return post<OrderVO>('/orders', data)
}

// 更新订单状态
export function updateOrderStatus(id: number, status: OrderStatus) {
  return put<OrderVO>(`/orders/${id}/status`, { status })
}

// 删除订单
export function deleteOrder(id: number) {
  return del<void>(`/orders/${id}`)
}
